### PR TITLE
7 - Improve and update sc documentation massa docs (1/3)

### PR DESCRIPTION
**Pull request by Thykof**
TODO:
- [x] update "Introduction to Smart Contracts" and rename it to "Getting started"
- [x] remove deprecated tic-tac-toe example
- [x] add sum smart-contract example
- [ ] fetch code snippets from github instead of hard code it (can be done by creating a custom RST directive https://docutils.sourceforge.io/docs/howto/rst-directives.html) (optional)
- [x] interact with the sum smart contract

### CODE REVIEW

File: web3-dev/smart-contracts/getting-started.rst [hunk 1]
```diff
@@ -2,41 +2,40 @@
...
+===============
 
 In this section you will learn how to compile your first Massa smart contract.
 
 Setting up a new project
 ------------------------
 
 Make sure you have a recent version of Node.js and npm. Update or `install <https://docs.npmjs.com/downloading-and-installing-node-js-and-npm>`_ them if needed.
-Create or go to the directory where you want to set up your project and run:
+`massa-sc-toolkit <https://github.com/massalabs/massa-sc-toolkit/>`_ is a tool that creates a boilerplate smart-contract project.
```
Comment (qdrn) [inline web3-dev/smart-contracts/getting-started.rst:14]:
```suggestion

`massa-sc-toolkit <https://github.com/massalabs/massa-sc-toolkit/>`_ is a tool that creates a boilerplate smart-contract project.
```

File: web3-dev/smart-contracts/getting-started.rst [hunk 2]
```diff
@@ -2,41 +2,40 @@
...
-   npx massa-sc-create my-sc
+  npx @massalabs/sc-toolkit init my-sc
 
 Now that the npm project is created, go inside your smart-contract directory and install the dependencies using the following commands:
 
 .. code-block:: shell
 
    cd my-sc
-   npm install --legacy-peer-deps
+   npm install
```
Comment (qdrn) [inline web3-dev/smart-contracts/getting-started.rst:26]: Is that still necessary ? I have the impression that massalabs/massa-sc-toolkit comes with the node_modules. In that case I would simply say to open the folder with your favorite IDE. Or remove this line altogether.

File: web3-dev/smart-contracts/getting-started.rst [hunk 3]
```diff
@@ -97,7 +96,7 @@ Let's go!
 Configure the client
 ~~~~~~~~~~~~~~~~~~~~
 
-Make sure that you have the last version of the Massa node. If not, `install it <https://github.com/massalabs/massa/wiki/install>`_.
+Make sure that you have the last version of the Massa node. If not, `install it <https://docs.massa.net/en/latest/testnet/install.html>`_ and `run it <https://docs.massa.net/en/latest/testnet/running.html>`_.
```
Comment (qdrn) [inline web3-dev/smart-contracts/getting-started.rst:99]:
```suggestion
Make sure that you have the latest version of the Massa node. If not, `install it <https://docs.massa.net/en/latest/testnet/install.html>`_ and `run it <https://docs.massa.net/en/latest/testnet/running.html>`_.
```

Comment (qdrn) [inline web3-dev/smart-contracts/getting-started.rst:99]: These links should probably replaced by relative links (check rst doc if needed/index.rst).

**Review by qdrn — CHANGES_REQUESTED** [review]

File: web3-dev/smart-contracts/getting-started.rst (hunk 2 above)
Comment (Thykof) [reply web3-dev/smart-contracts/getting-started.rst:27]: you're right, the toolkit performs the `npm install`

File: web3-dev/smart-contracts/smart-contract-example-sum.rst [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+In this tutorial, we will go through all the steps required to create a smart-contract on Massa.
+
+You can find the complete project on this `Github repository <https://github.com/massalabs/massa-sc-examples/tree/main/games/tictactoe>`__.
+
+Prerequisites
+-------------
+
+Smart-contracts are written in `Assembly Script <https://www.assemblyscript.org/>`_, and so we’ll assume that you have some familiarity with it, but you should be able to follow along even if you’re coming from a different programming language. We’ll also assume that you’re familiar with programming concepts like functions, objects, arrays, and to a lesser extent, classes.
+
+For the decentralized website part, we'll assume that you have some familiarity with HTML and JavaScript. If you want to have more details, you can follow `this great tutorial from React <https://reactjs.org/tutorial/tutorial.html>`_ from which the dApp is heavily inspired from.
```
[the remark this answers is not in the document]
Comment (Thykof) [reply web3-dev/smart-contracts/smart-contract-example-sum.rst:19]: Line 17 sounds good to me.

Line 19 is about front end app so yes not needed here at all, I will move it to the dapp section (https://github.com/massalabs/massa-docs/issues/12).

File: web3-dev/smart-contracts/smart-contract-example-sum.rst [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+
+    curl --location --request POST 'https://inno.massa.net/test15' \
+    --header 'Content-Type: application/json' \
+    --data-raw '{
+      "jsonrpc": "2.0",
+      "method": "get_filtered_sc_output_event",
+      "params": [
+        {
+          "start": {
+              "period":17740,
```
[the remark this answers is not in the document]
Comment (Thykof) [reply web3-dev/smart-contracts/smart-contract-example-sum.rst:219]: I will use operation id

File: web3-dev/smart-contracts/smart-contract-example-sum.rst [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+.. code-block::
+
+    npm run deploy build/run.wasm
+
+Remember that our sum smart-contract compute the sum and emit an event with the result.
+So we will now look for the emitted event using the node RPC API.
+
+.. code-block::
+
+    curl --location --request POST 'https://inno.massa.net/test15' \
```
[the remark this answers is not in the document]
Comment (Thykof) [reply web3-dev/smart-contracts/smart-contract-example-sum.rst:211]: I have updated this part

File: web3-dev/smart-contracts/smart-contract-example-sum.rst [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+
+    Contract address is :  A1PjpgXyXSBeiG1rbXCP4ybhVccYzpysDKYmkymXWd81idutaD9
+
+
+Interaction with the smart-contract
+-----------------------------------
+
+We will now interact with our sum smart-contract.
+
+To interact with a smart-contract, we need to write another smart-contract that will be executed.
```
[the remark this answers is not in the document]
Comment (Thykof) [reply web3-dev/smart-contracts/smart-contract-example-sum.rst:170]: I have updated this part

File: web3-dev/smart-contracts/smart-contract-example-sum.rst [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+    npm run build
+
+.. _sending-sc-sum:
+
+Putting your smart-contract on the blockchain
+---------------------------------------------
+
+We'll now turn to the process of putting the smart-contract on the Massa blockchain.
+
+For the deployment, you will need a wallet with some coins. Rename the file `.env.example` into `.env` and fill it with your wallet keys and address.
```
[the remark this answers is not in the document]
Comment (Thykof) [reply web3-dev/smart-contracts/smart-contract-example-sum.rst:134]: I have updated this part

File: web3-dev/smart-contracts/getting-started.rst (hunk 3 above)
Comment (Thykof) [reply web3-dev/smart-contracts/getting-started.rst]: I use ref

File: web3-dev/smart-contracts/getting-started.rst [hunk 9]
```diff
@@ -107,14 +106,18 @@ If you are using an existing wallet, make sure that you have some coins on it.
 
 In any case, keep the `address` of your wallet, you will use it later.
 
+.. note::
+
+   You can also execute your smart-contract on a local sandbox node. To learn more about sandbox node, follow this tutorial: `Local network generation <https://docs.massa.net/en/latest/web3-dev/dummy-network-generation.html>`_.
```
[the remark this answers is not in the document]
Comment (Thykof) [reply web3-dev/smart-contracts/getting-started.rst]: I use ref

File: web3-dev/smart-contracts/smart-contract-example-sum.rst [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+      return x;
+    }
+
+Calling function of a smart-contract that is stored in the blockchain with some arguments will start an assemblyscript runtime (wasmer).
+This is why each function that you want to be able to call in your smart-contract must be exported with the `export` keyword and must take one string argument and return a value of type string.
+
+Here, we are exporting the sum function. In this function, we deserialize the argument into two integers, with the help of `fromByteString` and `toInt32`.
+
+.. note::
+    Massalabs team is working on a better way to serialize and deserialize the function arguments.
```
[the remark this answers is not in the document]
Comment (Thykof) [reply web3-dev/smart-contracts/smart-contract-example-sum.rst:93]: We have a lot of trouble with json serialization, this is why we choose byte encoding

[the remark this answers is not in the document]
Comment (qdrn) [reply web3-dev/smart-contracts/smart-contract-example-sum.rst:93]: I know. But it still work and is probably easier to read in most cases so I believe it's important to mention it. On my side I've been using json without much problems (for probably simple use cases).

**Review by qdrn — APPROVED** [review]
Sounds good. I would still mention that json is an option (see comment). But we can start like this.

**Comment by Thykof** [comment]
@gregLibert If you want to have a look at this PR and share your thoughs, there is https://github.com/massalabs/massa-docs/issues/12 to continue improving the documentation.

File: web3-dev/dummy-network-generation.rst [hunk 11]
```diff
@@ -32,7 +34,7 @@ Create a keypair in massa-client:
     cd massa-client && cargo run
     wallet_generate_secret_key
 
-For the rest of the tutorial we will use theses abreviations:
+For the rest of the tutorial we will use theses abbreviations:
 
 - `SECRETK` : The secret you just generated
 - `PUBK` : The public key corresponding to SECRETK
```
[the remark this answers is not in the document]
Comment (qdrn) [reply web3-dev/dummy-network-generation.rst:40]: It's called secret key across all the massa codebase. I also wonder why it was changed.

File: web3-dev/smart-contracts/getting-started.rst [hunk 12]
```diff
@@ -2,41 +2,40 @@
...
 
-   npm install --global yarn npx
-   npx massa-sc-create my-sc
+  npx @massalabs/sc-toolkit init my-sc
 
 Now that the npm project is created, go inside your smart-contract directory and install the dependencies using the following commands:
 
 .. code-block:: shell
 
    cd my-sc
```
[the remark this answers is not in the document]
Comment (Thykof) [reply web3-dev/smart-contracts/getting-started.rst:26]: yes the toolkit does it https://github.com/massalabs/massa-sc-toolkit/blob/aeefae878c9cc16f054b219a5cc300d62ec50017/initialize.js#L26

**Pull request merged**